tests/cbmc/include/cbmc_proof/cbmc_utils.h

```diff
@@ -27,6 +27,11 @@
     do {                         \
         if (!(cond)) { return; } \
     } while (0)
+#define CBMC_ENSURE_REF_RETURN_NULL(cond)    \
+    do {                         \
```

Comment:
Would the s2n "[PTR_ENSURE_REF](https://github.com/aws/s2n-tls/blob/main/utils/s2n_safety_macros.h#L514)" method meet your needs? Or are you trying to avoid setting s2n_error?

Reply:
I was wondering the same thing

Reply:
The s2n error codes were not necessary in the CBMC collateral, but I think you're right that this is a duplication of effort.  I've updated the pull request.

I don't know the historical reasons for using `CBMC_ENSURE_REF` instead of using the s2n safety macros.  My guess is that the intention was to have a void return type, and the s2n safety macros return an int error code.  Eventually we might want to remove usage of `CBMC_ENSURE_REF` as well.

>  * DEPRECATED: all methods (except those in s2n.h) should return s2n_result.
Should we be concerned by the comment for `PTR_ENSURE_REF` stating that it is deprecated?

Reply:
No concern unless we eventually actually remove it, and then we can always put it somewhere the cbmc proofs can continue to access. It's just deprecated for s2n-tls code because none of our functions should return a pointer. We want all functions to return S2N_RESULT so we have a clear, consistent error signal.